orion/controllers/programs/programsController.go

```diff
@@ -81,6 +85,24 @@ func DeleteProgram(c *gin.Context) {
   return
 }
 
-func CheckValidProgram() bool {
-  return true
+func CheckValidProgram(program Program) error {
+  programId := program.ProgramId
+  grade1 := program.Grade1
+  grade2 := program.Grade2
+
+  if programId == "" {
+    return errors.New("empty Program ID")
+  }
+
+  // Checks if the program ID is in the form of alphanumeric strings separated by underscores
+  if matches, _ := regexp.MatchString("^[[:alnum:]]+(_[[:alnum:]]+)*$", programId); !matches {
```

Comment:
> "^[[:alnum:]]+(_[[:alnum:]]+)*$"

Can you make this a global variable?
Every time we call this function, we re-create this String.
Instead, with a global variable, it is created once and simply always referenced.

At the top of the file, you could do:
```
REGEX_PROGRAM_ID := "..."
```

I like all upper-cased and snake case here because it signals this variable should NOT be changed and it's a global variable. Variables with camelCase imply a "regular" variable.